File: sprint JAVA/desktopApplication/src/entity/pojet/Projet.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity.pojet;

import java.sql.Date;

/**
 *
 * @author Aymen Touihri
 */
public class Projet {

    public enum status {
        INPROGRESS,
        FINISHED,
        NOTSTARTED,
        ONHOLD;

        public static String enumToString(status pType) {
            return pType.toString();
        }

        public String enumToString() {
            return this.toString();
        }

        public static status stringToEnum(String pTypeStr) {
            switch (pTypeStr.toUpperCase()) {
                case "INPROGRESS":
                    return status.INPROGRESS;
                case "FINISHED":
                    return status.FINISHED;
                case "ONHOLD":
                    return status.ONHOLD;
                case "NOTSTARTED":
                    return status.NOTSTARTED;
                default:
                    return status.INPROGRESS;
            }
        }
    }
    private int idProjet;
    private String Nom;
    private Date dateDebut;
    private Date dateFin;
    private String description;
    private status Status;

    public Projet(int idProjet, String Nom, Date dateDebut, Date dateFin, String description, status Status) {
        this.idProjet = idProjet;
        this.Nom = Nom;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.description = description;
        this.Status = Status;
    }

     public Projet(int idProjet,String nom) {
        this.idProjet = idProjet;
        this.Nom = nom;
    }
     
    public Projet() {
    }

    public int getIdProjet() {
        return idProjet;
    }

    public status getStatus() {
        return Status;
    }

    public void setStatus(status Status) {
        this.Status = Status;
    }

    public String getNom() {
        return Nom;
    }

    public Date getDateDebut() {
        return dateDebut;
    }

    public Date getDateFin() {
        return dateFin;
    }

    public String getDescription() {
        return description;
    }

    public void setIdProjet(int idProjet) {
        this.idProjet = idProjet;
    }

    public void setNom(String Nom) {
        this.Nom = Nom;
    }

    public void setDateDebut(Date dateDebut) {
        this.dateDebut = dateDebut;
    }

    public void setDateFin(Date dateFin) {
        this.dateFin = dateFin;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "projet{" + "idProjet=" + idProjet + ", Nom=" + Nom + ", dateDebut=" + dateDebut + ", dateFin=" + dateFin + ", description=" + description + '}';
    }

}

File: sprint SYMFONY/webApplication/src/ProjetBundle/Entity/Projet.php
<?php

namespace ProjetBundle\Entity;

use DateTime;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * Projet
 *
 * @ORM\Table(name="projet")
 * @ORM\Entity(repositoryClass="ProjetBundle\Repository\ProjetRepository")
 */
class Projet implements \JsonSerializable
{
    /**
     * @var integer
     *
     * @ORM\Column(name="idProjet", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idprojet;

    /**
     * @var string
     * @Assert\NotBlank
     * @ORM\Column(name="nom", type="string", length=20, nullable=false)
     */
    private $nom;

    /**
     * @var DateTime
     * @Assert\Type(
     *      type = "\DateTime",
     *      message = "Start Date invalid ",
     * )

     * @ORM\Column(name="dateDebut", type="date", nullable=false)
     */
    private $datedebut;

    /**
     * @var DateTime
     * @Assert\Type(
     *      type = "\DateTime",
     *      message = "End Date invalid",
     * )
     * @Assert\GreaterThanOrEqual(
     *      value = "today",
     *      message = "Must be equal or greater than today date"
     * )
     * @Assert\Expression(
     *     "this.getDatefin() >= this.getDatedebut()",
     *     message="Must be greater than Start date .."
     * )
     * @ORM\Column(name="dateFin", type="date", nullable=false)
     */
    private $datefin;

    /**
     * @var string
     * @Assert\Length(
     *      min = 2,
     *      max = 50,
     *      minMessage = "Your first name must be at least {{ limit }} characters long",
     *      maxMessage = "Your first name cannot be longer than {{ limit }} characters"
     * )
     * @ORM\Column(name="description", type="text", length=65535, nullable=false)
     */
    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="status", type="string", length=20, nullable=false)
     */
    private $status='Notstarted';





    /**
     * @return int
     */
    public function getIdprojet()
    {
        return $this->idprojet;
    }

    /**
     * @param int $idprojet
     * @return Projet
     */
    public function setIdprojet($idprojet)
    {
        $this->idprojet = $idprojet;
        return $this;
    }

    /**
     * @return string
     */
    public function getNom()
    {
        return $this->nom;
    }

    /**
     * @param string $nom
     * @return Projet
     */
    public function setNom($nom)
    {
        $this->nom = $nom;
        return $this;
    }

    /**
     * @return DateTime
     */
    public function getDatedebut()
    {
        return $this->datedebut;
    }

    /**
     * @param DateTime $datedebut
     * @return Projet
     */
    public function setDatedebut($datedebut)
    {
        $this->datedebut = $datedebut;
        return $this;
    }

    /**
     * @return DateTime
     */
    public function getDatefin()
    {
        return $this->datefin;
    }

    /**
     * @param DateTime $datefin
     * @return Projet
     */
    public function setDatefin($datefin)
    {
        $this->datefin = $datefin;
        return $this;
    }

    /**
     * @return string
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * @param string $description
     * @return Projet
     */
    public function setDescription($description)
    {
        $this->description = $description;
        return $this;
    }

    /**
     * @return string
     */
    public function getStatus()
    {
        return $this->status;
    }

    /**
     * @param string $status
     * @return Projet
     */
    public function setStatus($status)
    {
        $this->status = $status;
        return $this;
    }

   

    public function __toString()
    {
        return  $this->nom;
    }

    /**
     * @inheritDoc
     */
    public function jsonSerialize()
    {
        return get_object_vars($this);
    }

}

